pino.js

```diff
@@ -338,6 +343,15 @@ function pino (opts, stream) {
     }
   }
 
+  var baseLog = (typeof iopts.baseLog === 'object') ? iopts.baseLog : defaultBaseLog
+  instance = instance.child(baseLog)
+
+  if (iopts.name !== undefined) {
+    instance = instance.child({
```

Comment:
rather than creating two children, we can just merge this into `baseLog` - this will avoid duplicate name keys